registry/api/v2/errors.go

```diff
@@ -154,4 +154,14 @@ var (
 		to return) is not an integer, or "n" is negative.`,
 		HTTPStatusCode: http.StatusBadRequest,
 	})
+
+	// ErrorCodeMalformedNextToken is returned when uploading a blob if the
+	// provided digest does not match the blob contents.
+	ErrorCodeMalformedNextToken = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
I've been giving this some thought myself, but we may want to introduce some way to register errors from the extension API itself.